mm2src/coins/lp_coins.rs

```diff
@@ -2150,36 +2161,106 @@ impl CoinsContext {
+        self.add_token(coin).await
+    }
+
     pub async fn add_platform_with_tokens(
         &self,
         platform: MmCoinEnum,
         tokens: Vec<MmCoinEnum>,
     ) -> Result<(), MmError<PlatformIsAlreadyActivatedErr>> {
         let mut coins = self.coins.lock().await;
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+
         if coins.contains_key(platform.ticker()) {
             return MmError::err(PlatformIsAlreadyActivatedErr {
                 ticker: platform.ticker().into(),
             });
         }
 
-        coins.insert(platform.ticker().into(), platform);
+        let platform_ticker = platform.ticker().to_string();
+        coins.insert(platform_ticker.clone(), platform);
 
         // Tokens can't be activated without platform coin so we can safely insert them without checking prior existence
+        let mut token_tickers = Vec::with_capacity(tokens.len());
+        // TODO
+        // Handling for these case:
+        // USDT was activated via enable RPC
+        // We try to activate ETH coin and USDT token via enable_eth_with_tokens
         for token in tokens {
+            token_tickers.push(token.ticker().to_string());
             coins.insert(token.ticker().into(), token);
         }
+
+        platform_coin_tokens
+            .entry(platform_ticker)
+            .or_default()
+            .extend(token_tickers);
+        Ok(())
+    }
+
+    /// Get enabled coins to disable.
+    pub async fn get_tokens_to_disable(&self, ticker: &str) -> HashSet<String> {
+        let coins = self.platform_coin_tokens.lock();
+        coins.get(ticker).cloned().unwrap_or_default()
+    }
+
+    pub async fn remove_coin(&self, coin: MmCoinEnum) -> Result<(), String> {
+        let ticker = coin.ticker();
+        let platform_ticker = coin.platform_ticker();
+        let mut coins_storage = self.coins.lock().await;
+        let mut platform_tokens_storage = self.platform_coin_tokens.lock();
+
+        // Check if ticker is a platform coin and remove from it platform's token list
+        if ticker == platform_ticker {
+            if let Some(tokens_to_remove) = platform_tokens_storage.remove(ticker) {
+                tokens_to_remove.iter().for_each(|token| {
+                    if let Some(token) = coins_storage.remove(token) {
+                        // Abort all token related futures on token deactivation
+                        token
+                            .on_disabled()
+                            .error_log_with_msg(&format!("Error aborting coin({ticker}) futures"));
+                    }
+                });
+            };
+        } else if let Some(tokens) = platform_tokens_storage.get_mut(platform_ticker) {
+            tokens.remove(ticker);
+            coin.on_token_deactivated(ticker);
+            coin.on_disabled()
+                .error_log_with_msg(&format!("Error aborting coin({ticker}) futures"));
+        };
```

Comment:
Just noticed two issues here:
1. If there is no `platform_ticker` in `platform_tokens_storage`, we won't call `coin.on_disabled`
This can be solved via:
```rust
if ticker == platform_ticker {
  ...
} else {
  if let Some(tokens) = platform_tokens_storage.get_mut(platform_ticker) {
    tokens.remove(ticker);
  }
  coin.on_disabled()
                .error_log_with_msg(&format!("Error aborting coin({ticker}) futures"));
  // Also here do the suggestion from the point 2
}
```
2. Here you produced the same error as I mentioned here: https://github.com/KomodoPlatform/atomicDEX-API/pull/1525#discussion_r1032429682